File: frontend/src/app/shared/service/admin/courseref/LevelTestConfigurationAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {LevelTestConfigurationDto} from 'src/app/shared/model/courseref/LevelTestConfiguration.model';
import {LevelTestConfigurationCriteria} from 'src/app/shared/criteria/courseref/LevelTestConfigurationCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class LevelTestConfigurationAdminService extends AbstractService<LevelTestConfigurationDto, LevelTestConfigurationCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): LevelTestConfigurationDto {
        return new LevelTestConfigurationDto();
    }

    public constrcutCriteria(): LevelTestConfigurationCriteria {
        return new LevelTestConfigurationCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/levelTestConfiguration/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/courseref/LevelTestConfigurationAdminService.java
package ma.zs.alc.service.facade.admin.courseref;

import java.util.List;
import ma.zs.alc.bean.core.courseref.LevelTestConfiguration;
import ma.zs.alc.dao.criteria.core.courseref.LevelTestConfigurationCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface LevelTestConfigurationAdminService extends  IService<LevelTestConfiguration,LevelTestConfigurationCriteria>  {

    List<LevelTestConfiguration> findByParcoursId(Long id);
    int deleteByParcoursId(Long id);
    long countByParcoursCode(String code);




}
